pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -337,10 +372,12 @@ func (m swapConfigurationHelper) ConfigureLimitedSwap(lcr *runtimeapi.LinuxConta
 
 func (m swapConfigurationHelper) ConfigureNoSwap(lcr *runtimeapi.LinuxContainerResources) {
 	if !isCgroup2UnifiedMode() {
-		// memorySwapLimit = total permitted memory+swap; if equal to memory limit, => 0 swap above memory limit
-		// Some swapping is still possible.
-		// Note that if memory limit is 0, memory swap limit is ignored.
-		lcr.MemorySwapLimitInBytes = lcr.MemoryLimitInBytes
+		if swapControllerAvailable() {
```

Comment:
Question: Should this not be checked regardless of whether unified mode is enabled. Otherwise we explicitly set `memory.swap.max` even if swap is disabled?

For reference, I am having issues launching a >= 1.28.0 using `kind` on a node where swap is disabled. (I will spend some time creating an issue tomorrow).

cc @iholder101

Reply:
Is this fix a problem? I cannot catch up with you here.

This added a check for `no-cgroup v2`: to not set swap if swap controller is not available.

Reply:
I think this fix is fine. I was asking whether this check should also be done on `cgroup v2` systems. I have a PR #120784 under review to do this -- effectively moving the check higher up the call chain.

Feel free to review this as well.